### PR TITLE
Feat/advanced fields ast

### PR DESCRIPTION
**Pull request by yyyyaaa**
This PR adds some custom ast resolving logic for launchql-client, I'll keep updating this PR as I'm making changes and exploring the other types on client side

### CODE REVIEW

**Comment by pyramation** [comment]
super cool! 
I saw the interval changed from `$timeRequired: BigFloat` to `$timeRequired: [object Object]`

I think we patch/insert with float, but we query with the object? I forgot but just wanted to mention in case

**Comment by yyyyaaa** [comment]
> super cool!  awesome work @yyyyaaa ... this is looking good and stoked that we're making custom types
> 
> I saw the interval changed from `$timeRequired: BigFloat` to `$timeRequired: [object Object]`
> 
> I think we patch/insert with float, but we query with the object? I forgot but just wanted to mention in case 

Hmm it could be that I haven't updated pg, I saw it was object in the schema

File: packages/launchql-client/src/custom-ast.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  return t.field({
+    name: fieldDefn.name
+  });
+}
+
+export function geometryPointAst(name) {
+  return t.field({
+    name,
+    selectionSet: t.selectionSet({
+      selections: toFieldArray(['geojson', 'srid', 'x', 'y'])
```
Comment (pyramation) [inline packages/launchql-client/src/custom-ast.js:22]: nice!!!!

File: packages/launchql-client/src/custom-ast.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import * as t from 'gql-ast';
+
+export function getCustomAst(fieldDefn) {
+  const { subtype, pgType } = fieldDefn.type;
+  if (subtype === 'GeometryPoint' && pgType === 'geometry') {
+    return geometryPointAst(fieldDefn.name);
```
Comment (pyramation) [inline packages/launchql-client/src/custom-ast.js:6]: 🚀

**Comment by yyyyaaa** [comment]
@pyramation fixed a few more bugs, could you review for me?

**Comment by yyyyaaa** [comment]
mainly in mutations, for example, previously it was:
```
mutation updateActionBySlugMutation(
  $slug: String!
  $photo: JSON
  $title: String
  $url: String
  $description: String
  $discoveryHeader: String
  $discoveryDescription: String
  $enableNotifications: Boolean
  $enableNotificationsText: String
  $search: FullText
  $location: GeoJSON
  $locationRadius: BigFloat
  $timeRequired: [Object Object]
  $startDate: Datetime
  $endDate: Datetime
  $approved: Boolean
  $rewardAmount: BigFloat
  $activityFeedText: String
  $callToAction: String
  $completedActionText: String
  $alreadyCompletedActionText: String
  $tags: [String]
  $ownerId: UUID
  $photoUpload: Upload
) {
  updateActionBySlug(
    input: {
      slug: $slug
      patch: {
        photo: $photo
        title: $title
        url: $url
        description: $description
        discoveryHeader: $discoveryHeader
        discoveryDescription: $discoveryDescription
        enableNotifications: $enableNotifications
        enableNotificationsText: $enableNotificationsText
        search: $search
        location: $location
        locationRadius: $locationRadius
        timeRequired: $timeRequired
        startDate: $startDate
        endDate: $endDate
        approved: $approved
        rewardAmount: $rewardAmount
        activityFeedText: $activityFeedText
        callToAction: $callToAction
        completedActionText: $completedActionText
        alreadyCompletedActionText: $alreadyCompletedActionText
        tags: $tags
        ownerId: $ownerId
        photoUpload: $photoUpload
      }
    }
  ) {
    action {
      id
      slug
      photo
      title
      url
      description
      discoveryHeader
      discoveryDescription
      enableNotifications
      enableNotificationsText
      search
      locationRadius
      startDate
      endDate
      approved
      rewardAmount
      activityFeedText
      callToAction
      completedActionText
      alreadyCompletedActionText
      tags
      createdBy
      updatedBy
      createdAt
      updatedAt
    }
  }
}

```

Note the `[Object Object]` of `timeRequired`, previous code did not handle if the input attribute has `properties` key

**Comment by yyyyaaa** [comment]
oh actually I have just discover 1 more edge case, WIP now til the fix

**Comment by yyyyaaa** [comment]
all fixed

**Pull request merged**